### PR TITLE
[Docs] Add a document about setting interval

### PR DESCRIPTION
**Pull request by YuetianW**
Add a document about Setting the Frequency of Logging, Weight Saving, and Validation.

### CODE REVIEW

**Comment by HAOCHENYE** [comment]
Hi! Thanks for your contributions! Although we've updated a document to tell users how to use epoch-based training and iteration-based training, I think the tutorial you provide has a much more detailed description that can help users to understand why and how to set up epoch-based or iteration-based training. Would mind merging this tutorial into this one?

**Comment by YuetianW** [comment]
Hello! Thank you for your reply!
It would be my pleasure if my PR could be merged.

I hope that mmengine will be perfected step by step under the joint efforts of everyone！






王悦天
***@***.***

华中科技大学


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "open-mmlab/mmengine"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月25日(星期六) 凌晨1:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [open-mmlab/mmengine] [Docs]Add a document about setting interval. (PR #959)





 
Hi! Thanks for your contributions! Although we've updated a document to tell users how to use epoch-based training and iteration-based training, I think the tutorial you provide has a much more detailed description that can help users to understand why and how to set up epoch-based or iteration-based training. Would mind merging this tutorial into this one?

**Comment by zhouzaida** [comment]
> Hello! Thank you for your reply! It would be my pleasure if my PR could be merged. I hope that mmengine will be perfected step by step under the joint efforts of everyone！

Hi, thanks for your contribution. Let us make it better. We will review this PR ASAP.

**Comment by YuetianW** [comment]
This PR link issue #962 .

**Comment by zhouzaida** [comment]
Hi, the lint failed. You can refer to [CONTRIBUTING.md#2-configure-pre-commit](https://github.com/open-mmlab/mmengine/blob/main/CONTRIBUTING.md#2-configure-pre-commit) to fix it.

**Comment by YuetianW** [comment]
> Hi, the lint failed. You can refer to [CONTRIBUTING.md#2-configure-pre-commit](https://github.com/open-mmlab/mmengine/blob/main/CONTRIBUTING.md#2-configure-pre-commit) to fix it.

22/5000
Thanks for your reminder, it has been fixed by pre-commit.

File: docs/en/common_usage/set_interval.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Setting the Frequency of Logging, Weight Saving, and Validation
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:1]:
```suggestion
# Setting the Frequency of Logging, Checkpoint Saving, and Validation
```

File: docs/en/common_usage/set_interval.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+MMEngine supports two training modes, `EpochBased` based on epochs and `IterBased` based on the number of iterations. Both of these modes are used in downstream algorithm libraries such as [MMDetection](https://github.com/open-mmlab/mmdetection), which uses the `EpochBased` mode by default, and [MMSegmentation](https://github.com/open-mmlab/mmsegmentation), which uses the `IterBased` mode by default.
+
+Under different training modes, the semantics of the interval in MMEngine will be different. The interval in `EpochBased` mode is based on epochs, while that in `IterBased` mode is based on iterations.
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:5]:
```suggestion
In different training modes, the semantics of the interval in MMEngine will be different. In EpochBased mode, the interval is in terms of epochs, while in IterBased mode, the interval is in terms of iterations.
```

File: docs/en/common_usage/set_interval.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+## Setting the Interval for Training and Validation
+
+To customize the interval for training and validation, set the `val_interval` parameter in the initialization parameter `train_cfg` of [Runner](mmengine.runner.Runner).
+
+- EpochBased
+
+In `EpochBased` mode, the default value of `val_interval` is 1, which means to validate once after training an epoch.
+
+```
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:15]:
```suggestion
```python
```

File: docs/en/common_usage/set_interval.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    val_evaluator=dict(type=Accuracy),
+)
+runner.train()
+```
+
+- IterBased
+
+In `IterBased` mode, the default value of `val_interval` is 1000, which means to validate once after training 1000 iterations.
+
+```
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:33]:
```suggestion
```python
```

File: docs/en/common_usage/set_interval.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    optim_wrapper=dict(optimizer=dict(type=SGD, lr=0.001, momentum=0.9)),
+    train_cfg=dict(by_epoch=False, max_iters=10000, val_interval=2000),
+    val_dataloader=val_dataloader,
+    val_cfg=dict(),
+    val_evaluator=dict(type=Accuracy),
+)
+runner.train()
+```
+
+## Setting the Interval for Saving Weights
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:47]:
```suggestion
## Setting the Interval for Saving Checkpoints
```

File: docs/en/common_usage/set_interval.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+To customize the interval for saving weights, set the `interval` parameter of [CheckpointHook](mmengine.hooks.CheckpointHook).
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:49]:
```suggestion
To customize the interval for saving checkpoints, set the `interval` parameter of [CheckpointHook](mmengine.hooks.CheckpointHook).
```

File: docs/en/common_usage/set_interval.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+- EpochBased
+
+In `EpochBased` mode, the default value of `interval` is 1, which means saving weights once after training for one epoch.
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:53]:
```suggestion
In `EpochBased` mode, the default value of `interval` is 1, which means saving checkpoints once after training for one epoch.
```

File: docs/en/common_usage/set_interval.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+```
+# Set interval to 2, which means to save weights once every 2 epochs
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:56]:
```suggestion
```python
# Set the interval to 2, which means to save checkpoints every 2 epochs
```

File: docs/en/common_usage/set_interval.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    val_cfg=dict(),
+    val_evaluator=dict(type=Accuracy),
+    default_hooks=default_hooks,
+)
+runner.train()
+```
+
+- IterBased
+
+By default, weights are saved in units of epochs. If you want to save weights in units of iterations, you need to set `by_epoch=False`.
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:74]:
```suggestion
By default, checkpoints are saved in terms of epochs. If you want to save checkpoints in terms of iterations, you need to set `by_epoch=False`.
```

File: docs/en/common_usage/set_interval.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+```
+# Set by_epoch=False and interval=500, which means to save weights once every 500 iterations
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:77]:
```suggestion
```python
# set by_epoch=False and interval=500, which means to save checkpoints every 500 iterations
```

File: docs/en/common_usage/set_interval.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    val_cfg=dict(),
+    val_evaluator=dict(type=Accuracy),
+    default_hooks=default_hooks,
+)
+runner.train()
+```
+
+For more information on how to use `CheckpointHook`, please refer to the [CheckpointHook tutorial](../tutorials/hook.md#checkpointhook).
+
+## Setting the interval for printing logs
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:95]:
```suggestion
## Setting the Interval for Printing Logs
```

File: docs/en/common_usage/set_interval.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+By default, logs are printed to the terminal once every 10 iterations. You can set the interval using the `interval` parameter of the [LoggerHook](mmengine.hooks.LoggerHook).
+
+```
+pythonCopy code# Print logs every 20 iterations
```
Comment (zhouzaida) [inline docs/en/common_usage/set_interval.md:100]:
```suggestion
```python
# print logs every 20 iterations
```

**Comment by zhouzaida** [comment]
Hi, please also refer to #917 to update the file `en/zh_cn/index.rst`.

**Comment by YuetianW** [comment]
> Hi, please also refer to #917 to update the file `en/zh_cn/index.rst`.

en/index.rst is updated in the latest commit!

**Pull request closed without merging**